### PR TITLE
Added unit test for pkg/cloud/services/secretsmanager

### PR DESCRIPTION
**Pull request by Madhur97**
**What type of PR is this?**
/area testing

**What this PR does / why we need it**:
This PR increases the unit tests coverage for pkg/cloud/services/secretmanager package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2910

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Add unit tests for `pkg/cloud/services/secretmanager` package
```

### CODE REVIEW

File: pkg/cloud/services/secretsmanager/secret_test.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+	}
+}
+
+func setupClient() client.Client {
+	scheme := runtime.NewScheme()
+	_ = infrav1.AddToScheme(scheme)
+	return fake.NewClientBuilder().WithScheme(scheme).Build()
+}
+
+func getClusterScope(client client.Client) (*scope.ClusterScope, error) {
```
Comment (sedefsavas) [inline pkg/cloud/services/secretsmanager/secret_test.go:266]: nit. Duplicated code with other services.

File: pkg/cloud/services/secretsmanager/secret_test.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+		},
+	}
+	return scope.NewClusterScope(scope.ClusterScopeParams{
+		Client:     client,
+		Cluster:    cluster,
+		AWSCluster: &infrav1.AWSCluster{},
+	})
+}
+
+func getMachineScope(client client.Client, clusterScope *scope.ClusterScope) (*scope.MachineScope, error) {
```
Comment (sedefsavas) [inline pkg/cloud/services/secretsmanager/secret_test.go:279]: nit. Duplicated code with other services.

File: pkg/cloud/services/secretsmanager/secret_test.go [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+	tests := []struct {
+		name           string
+		bytesCount     int64
+		secretPrefix   string
+		expectedPrefix string
+		wantErr        bool
+		expect         func(g *WithT, m *mock_secretsmanageriface.MockSecretsManagerAPIMockRecorder)
+	}{
+		{
+			name:           "Should not store data in secret manager if data is having zero bytes",
```
Comment (sedefsavas) [inline pkg/cloud/services/secretsmanager/secret_test.go:94]: How is this scenario checked here by just checking nil error?

**Review by sedefsavas — COMMENTED** [review]
Regarding the duplication comments: we can do a deduplication PR after all tests are merged.

File: pkg/cloud/services/secretsmanager/secret_test.go (hunk 3 above)
Comment (Madhur97) [reply pkg/cloud/services/secretsmanager/secret_test.go:94]: The expect method of this test is nil, which means this test will succeed if there is no call to CreateSecret() method. If there is any call to CreateSecret(), the test will fail.
For other tests where we are expecting to store data in secret manager, we have set expected calls to CreateSecret() method.

I will add an expect method having `m.CreateSecret(gomock.Any()).Times(0)` to this test to make this more explicit.

File: pkg/cloud/services/secretsmanager/secret_test.go (hunk 1 above)
Comment (Madhur97) [reply pkg/cloud/services/secretsmanager/secret_test.go:266]: As suggested in other comment, we can do this after other related PRs are also merged.

File: pkg/cloud/services/secretsmanager/secret_test.go (hunk 3 above)
Comment (sedefsavas) [reply pkg/cloud/services/secretsmanager/secret_test.go:94]: No, this is enough. Thanks.

**Pull request merged**